algorithmic_efficiency/workloads/librispeech_conformer/librispeech_pytorch/model.py

```diff
@@ -0,0 +1,424 @@
+    pad_top = pad_along_height // 2
+    pad_bottom = pad_along_height - pad_top
+    pad_left = pad_along_width // 2
+    pad_right = pad_along_width - pad_left
+    return (pad_left, pad_right, pad_top, pad_bottom)
+
+  def forward(self, inputs, paddings):
+    groups = inputs.shape[1] // self.input_channels
+
+    if self.padding == 'SAME':
+      in_ = F.pad(inputs, self.get_same_padding(inputs.shape))
+    else:
+      in_ = inputs
+    outputs = F.conv2d(
+        input=in_,
+        weight=self.kernel,
+        bias=self.bias,
+        stride=self.filter_stride,
+        dilation=(1, 1),
+        groups=groups)
+
+    outputs = F.relu(outputs)
+
+    input_length = paddings.shape[1]
+    stride = self.filter_stride[0]
+    pad_len = (input_length + stride - 1) // stride * stride - input_length
+    out_padding = F.conv1d(
+        input=torch.cat([
+            paddings[:, None, :],
+            torch.zeros(
+                size=(paddings.shape[0], 1, pad_len), device=paddings.device)
+        ],
+                        dim=2),
+        weight=torch.ones([1, 1, 1], device=paddings.device),
+        stride=self.filter_stride[:1])
+    out_padding = out_padding.squeeze(dim=1)
+    outputs = outputs * (1 - out_padding[:, None, :, None])
+    return outputs, out_padding
+
+
+class FeedForwardModule(nn.Module):
+
+  def __init__(self, config: ConformerConfig):
+    super().__init__()
+    self.config = config
+
+    self.ln = LayerNorm(dim=config.encoder_dim)
+    self.linear1 = nn.LazyLinear(
+        out_features=config.encoder_dim * config.feed_forward_expansion_factor,
+        bias=True)
+    self.dropout1 = nn.Dropout(p=config.feed_forward_dropout_rate)
+    self.linear2 = nn.LazyLinear(out_features=config.encoder_dim, bias=True)
+    self.dropout2 = nn.Dropout(p=config.feed_forward_residual_dropout_rate)
+
+  def forward(self, inputs, padding_mask):
+    inputs = self.ln(inputs)
+    inputs = self.linear1(inputs)
+    inputs = F.silu(inputs)
+    inputs = self.dropout1(inputs)
+    inputs = inputs * padding_mask
+    inputs = self.linear2(inputs)
+    inputs = inputs * padding_mask
+    inputs = self.dropout2(inputs)
+
+    return inputs
+
+
+class AddPositionalEmbedding(nn.Module):
+
+  def __init__(self,
+               min_timescale: int = 1,
+               max_timescale: int = 10_000,
+               embedding_dim: int = 512):
+    super().__init__()
+    self.min_timescale = min_timescale
+    self.max_timescale = max_timescale
+    self.embedding_dim = embedding_dim
+    num_timescales = self.embedding_dim // 2
+    log_timescale_increment = math.log(
+        float(self.max_timescale) / float(self.min_timescale)) / (
+            num_timescales - 1)
+    inv_timescales = self.min_timescale * \
+        torch.exp(torch.arange(num_timescales, dtype=torch.float32)
+                  * -log_timescale_increment)
+    self.register_buffer("inv_timescales", inv_timescales[None, None, :])
```

Comment:
nit: here and elsewhere, use single quotes just so we're consistent with the rest of the codebase

Reply:
Done.